src/templater.rs

```diff
@@ -569,17 +569,12 @@ impl<'a> CommitOrChangeId<'a> {
 
     /// The length of the id printed will be the maximum of `total_len` and the
     /// length of the shortest unique prefix
-    pub fn shortest(&self, total_len: i64) -> ShortestIdPrefix {
-        let hex = self.hex();
-        let (prefix, rest) = extract_entire_prefix_and_trimmed_tail(
-            &hex,
-            self.repo.shortest_unique_id_prefix_len(self.as_bytes()),
-            max(total_len, 0) as usize,
-        );
-        ShortestIdPrefix {
-            prefix: prefix.to_string(),
-            rest: rest.to_string(),
-        }
+    pub fn shortest(&self, total_len: usize) -> ShortestIdPrefix {
+        let prefix_len = self.repo.shortest_unique_id_prefix_len(self.as_bytes());
+        let mut prefix = self.hex();
+        prefix.truncate(max(prefix_len, total_len));
```

Comment:
Perhaps add a comment here:

// When refactoring, note that it is theoretically possible that `prefix_len = prefix.len() + 1`.

Reply:
The comment gets less confusing if `prefix` is renamed to `hex`.

Reply:
Oh, I see what's your point. `prefix_len` can exceed the `hex.len()` if the `hex` were ambiguous. Then, the use of `.split_off()` is wrong, and need to clamp the value.

Good catch!